File: 1 - Orientação a objeto/class/veiculo.py
from abc import abstractmethod
from dataclasses import dataclass, field
from datetime import datetime
from typing import Dict

@dataclass( frozen=True)
class Veiculo():
    velocidade: int = field(init=True, default=None)
    pk: int = field(init=True, default=None)
    modelo: str = field(init=True, default=None)
    cor: str = field(init=True, default=None)

class VeiculoRepositorio():
    def criarVeiculo(params: Dict):
        veiculo = Veiculo(**params)
        return veiculo

File: 1 - Orientação a objeto/class/veiculo.ts
interface Iveiculo {
  _id?: Number ;
  velocidade: number;
  modelo: String;
  cor: String;
}

class Veiculo {
  _id: Number;
  velocidade: number;
  modelo: String;
  cor: String;

  constructor(params: Iveiculo) {
    this.velocidade = params.velocidade;
    this.modelo = params.modelo;
    this.cor = params.cor;
  }

  get id() {
    return this._id;
  }
}

export default Veiculo;
